daemon/graphdriver/quota/projectquota.go

```diff
@@ -153,8 +153,9 @@ func NewControl(basePath string) (*Control, error) {
 // SetQuota - assign a unique project id to directory and set the quota limits
 // for that project id
 func (q *Control) SetQuota(targetPath string, quota Quota) error {
-
+	q.RLock()
 	projectID, ok := q.quotas[targetPath]
+	q.Unlock()
 	if !ok {
 		projectID = q.nextProjectID
```

Comment:
Does this need a lock too?

Reply:
Looks like it... the code was written without concurrency in mind.

Reply:
maybe we could use atomics for this one

Reply:
yeah, it's easier to reuse the same lock